Tests/Context/ContextDecoratorPriorityQueueTest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Markup\NeedleBundle\Tests\Context;
+
+use Markup\NeedleBundle\Context\ContextDecoratorPriorityQueue;
+
+class ContextDecoratorPriorityQueueTest extends \PHPUnit_Framework_TestCase
+{
+    public function testAddingWithoutOrder()
+    {
+        $queue = new ContextDecoratorPriorityQueue();
+        $queue->insert('first', 100);
+        $queue->insert('second', 100);
+        $queue->insert('third', 100);
+        $queue->insert('fourth', 100);
+
+        $this->assertEquals('first|second|third|fourth', $this->implodeQueue($queue));
+    }
+
+    public function testOrder()
+    {
+        $queue = new ContextDecoratorPriorityQueue();
+        $queue->insert('fourth', -44);
+        $queue->insert('second', 100);
+        $queue->insert('first', 150);
+        $queue->insert('third', 99);
+
+        $this->assertEquals('first|second|third|fourth', $this->implodeQueue($queue));
+    }
+
```

Comment:
Its a required field http://php.net/manual/en/splpriorityqueue.insert.php I can't not give a priority